java/arcs/android/storage/database/DatabaseImpl.kt

```diff
@@ -395,14 +395,26 @@ class DatabaseImpl(
                 ) {
                     val rawSingletons = mutableMapOf<FieldName, Referencable?>()
                     val rawCollections = mutableMapOf<FieldName, Set<Referencable>>()
+                    val inlineStorageKeyId = it.getLong(3)
+                    val entityId = db.rawQuery(
+                        """
+                            SELECT
+                                entity_id
+                            FROM entities
+                            WHERE storage_key_id = ?
+                        """.trimIndent(),
+                        arrayOf(inlineStorageKeyId.toString())
+                    ).forSingleResult {
+                        it.getString(0)
+                    }
                     val (dbSingletons, dbCollections) =
-                        getEntityFields(it.getLong(3), counters, db)
+                        getEntityFields(inlineStorageKeyId, counters, db)
                     dbSingletons.forEach { (fieldName, value) -> rawSingletons[fieldName] = value }
                     dbCollections.forEach {
                         (fieldName, value) -> rawCollections[fieldName] = value
                     }
                     RawEntity(
-                        id = "",
+                        id = entityId!!,
```

Comment:
Prefer `checkNotNull(..)` or `requireNotNull(..)` to `!!`.   The former throws "IllegalStateException" and the latter throws "IllegalArgumentException"... which are both more helpful than the "NullPointerException" thrown by `!!` --- not to mention you can supply a message.